### PR TITLE
update churn

### PR DESCRIPTION
**Pull request by goern**
- address several RHBA in ubi8 python38 container image, update python dependencies
- address several RHBA in ubi8 python39 container image, update python dependencies
- address several RHBA in ubi8 python36 container image, update python dependencies
- python dependencies for f34 and f35

### CODE REVIEW

**Comment by goern** [comment]
is this superseding https://github.com/thoth-station/s2i-thoth/pull/237 ?

**Comment by mayaCostantini** [comment]
/lgtm
The new version of Thamos has been released so I think it should be good to go.

File: f34-py39/requirements.txt [hunk 1]
```diff
@@ -602,50 +551,50 @@ ruamel.yaml.clib==0.2.6; platform_python_implementation == 'CPython' and python_
...
+    --hash=sha256:782ef48d58982ddb49920c11a0c5c9c0b02e7d7d1c2ad0aa44e1a1e133051c96
 six==1.16.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2' \
     --hash=sha256:1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926 \
     --hash=sha256:8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254
 termcolor==1.1.0 \
     --hash=sha256:1d6d69ce66211143803fbc56652b41d73b4a400a2891d7bf7a1cdf4c02de613b
-thamos==1.21.1 \
-    --hash=sha256:38909ebb6345d98883fe03fd66e5204c014fafedff177662b6744b0c971467c9 \
-    --hash=sha256:b14907c7ead2a9bbeca0e80dc9a40d3c2152d7886ec5bbe9d605535cbdbf474a
+thamos==1.27.3 \
```
Comment (mayaCostantini) [inline f34-py39/requirements.txt:568]:
```suggestion
thamos==1.27.4 \
```

File: f35-py310/requirements.txt [hunk 2]
```diff
@@ -635,83 +551,51 @@ ruamel-yaml-clib==0.2.6 \
...
 termcolor==1.1.0 \
     --hash=sha256:1d6d69ce66211143803fbc56652b41d73b4a400a2891d7bf7a1cdf4c02de613b
-    # via
-    #   thamos
-    #   yaspin
-thamos==1.27.2 \
-    --hash=sha256:1806a6c481af397c1334edbc5870084abe3d37b40bb95036db32812cf056d85b \
-    --hash=sha256:7629cd878c96e20d0cee28678acdb1d5cac4f910f0d1a6654814650a80ad65c0
-    # via -r requirements.in
+thamos==1.27.3 \
```
Comment (mayaCostantini) [inline f35-py310/requirements.txt:568]:
```suggestion
thamos==1.27.4 \
```

File: ubi8-py38/requirements.txt [hunk 3]
```diff
@@ -622,50 +571,50 @@ ruamel.yaml.clib==0.2.6; platform_python_implementation == 'CPython' and python_
...
+    --hash=sha256:782ef48d58982ddb49920c11a0c5c9c0b02e7d7d1c2ad0aa44e1a1e133051c96
 six==1.16.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2' \
     --hash=sha256:1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926 \
     --hash=sha256:8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254
 termcolor==1.1.0 \
     --hash=sha256:1d6d69ce66211143803fbc56652b41d73b4a400a2891d7bf7a1cdf4c02de613b
-thamos==1.21.1 \
-    --hash=sha256:38909ebb6345d98883fe03fd66e5204c014fafedff177662b6744b0c971467c9 \
-    --hash=sha256:b14907c7ead2a9bbeca0e80dc9a40d3c2152d7886ec5bbe9d605535cbdbf474a
+thamos==1.27.3 \
```
Comment (mayaCostantini) [inline ubi8-py38/requirements.txt:588]:
```suggestion
thamos==1.27.4 \
```

File: ubi8-py39/requirements.txt [hunk 4]
```diff
@@ -602,50 +551,50 @@ ruamel.yaml.clib==0.2.6; platform_python_implementation == 'CPython' and python_
...
+    --hash=sha256:782ef48d58982ddb49920c11a0c5c9c0b02e7d7d1c2ad0aa44e1a1e133051c96
 six==1.16.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2' \
     --hash=sha256:1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926 \
     --hash=sha256:8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254
 termcolor==1.1.0 \
     --hash=sha256:1d6d69ce66211143803fbc56652b41d73b4a400a2891d7bf7a1cdf4c02de613b
-thamos==1.21.1 \
-    --hash=sha256:38909ebb6345d98883fe03fd66e5204c014fafedff177662b6744b0c971467c9 \
-    --hash=sha256:b14907c7ead2a9bbeca0e80dc9a40d3c2152d7886ec5bbe9d605535cbdbf474a
+thamos==1.27.3 \
```
Comment (mayaCostantini) [inline ubi8-py39/requirements.txt:568]:
```suggestion
thamos==1.27.4 \
```

**Comment by Gregory-Pereira** [comment]
In addition to maya's comments, ubi8-py36 is still using thamos version `1.21.1`

**Comment by Gregory-Pereira** [comment]
I tested each branch from your fork, not sure if I am building them wrong but they don't have access to the `thamos` cli, not sure if it is needed...
```
$ podman build -f f34-py39/Dockerfile -t f34-py39;
$ podman build -f f35-py310/Dockerfile -t f35-py310;
$ podman build -f ubi8-py38/Dockerfile -t ubi8-py38;
$ podman build -f ubi8-py39/Dockerfile -t ubi8-py39;
$ podman build -f ubi8-py36/Dockerfile -t ubi8-py36;
 
$ podman run --rm -it <each image tag> bash
(app-root) which thamos
bash: thamos: command not found
(app-root) pip list
Package    Version
---------- -------
pip        21.0.1
setuptools 39.2.0
wheel      0.31.1
```

File: f34-py39/requirements.txt [hunk 5]
```diff
@@ -602,50 +551,50 @@ ruamel.yaml.clib==0.2.6; platform_python_implementation == 'CPython' and python_
...
     --hash=sha256:1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926 \
     --hash=sha256:8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254
 termcolor==1.1.0 \
     --hash=sha256:1d6d69ce66211143803fbc56652b41d73b4a400a2891d7bf7a1cdf4c02de613b
-thamos==1.21.1 \
-    --hash=sha256:38909ebb6345d98883fe03fd66e5204c014fafedff177662b6744b0c971467c9 \
-    --hash=sha256:b14907c7ead2a9bbeca0e80dc9a40d3c2152d7886ec5bbe9d605535cbdbf474a
+thamos==1.27.4 \
+    --hash=sha256:71fe3fd6660b996c1276425ffad09dc9f1f2d608a34f75507019805aa77f1b5d \
+    --hash=sha256:98cc133c1977afca29ffc76314e4baacb6fa2463f3d607ca795bcb8bdbd311d5
```
Comment (Gregory-Pereira) [inline f34-py39/requirements.txt:570]: These are the wrong hashes (The old hashes are for thamos v1.27.3). I have provided the correct ones here.
```suggestion
    --hash=sha256:8def2131b43f0a6fb157f80dfb277ebfce387155ffb7e3da86b12ba5eef2be44 \
    --hash=sha256:d5d69c900b8ca90ff2feb9b89d358abddfbfb2c580359966206b93253a3019a2
```
The first hash on line 569 is the `.whl` and line 570 is the `.tar.gz`, and the same order has been kept for my suggestion

**Comment by Gregory-Pereira** [comment]
The `f34-py39` overlay is looking for `pip3` binary at path `/usr/pin/pip3`. Either use its current path `/opt/app-root/bin/pip3` or copy it to `/usr/bin`. This entails modifying [these two lines](https://github.com/goern/s2i-thoth/blob/update-churn/f34-py39/Dockerfile#L34-L35) in your fork. This also causes build to fail, but after this change and the thamos hash, it builds successfully.

**Comment by Gregory-Pereira** [comment]
Same changes apply to the `f35-py310` overlay as the `f34-py39` overlay.
    1. change [these lines](https://github.com/goern/s2i-thoth/blob/update-churn/f35-py310/Dockerfile#L34-L35) to use `/opt/app-root/bin/pip3` instead of `/usr/bin/pip3`.
    2. change [these lines](https://github.com/goern/s2i-thoth/blob/update-churn/f35-py310/requirements.txt#L569-L570) to use the correct hashes:
    ```
    --hash=sha256:8def2131b43f0a6fb157f80dfb277ebfce387155ffb7e3da86b12ba5eef2be44 \
    --hash=sha256:d5d69c900b8ca90ff2feb9b89d358abddfbfb2c580359966206b93253a3019a2
    ```
After these changes the `f35-py310` overlay builds successfully.

**Comment by Gregory-Pereira** [comment]
Same changes apply to the `ubi8-py38` overlay as the `f34-py39` overlay.
    1. change [these lines](https://github.com/goern/s2i-thoth/blob/update-churn/ubi8-py38/Dockerfile#L34-L35) to use `/opt/app-root/bin/pip3` instead of `/usr/bin/pip3`.
    2. change [these lines](https://github.com/goern/s2i-thoth/blob/update-churn/ubi8-py38/requirements.txt#L569-L570) to use the correct hashes:
```
    --hash=sha256:8def2131b43f0a6fb157f80dfb277ebfce387155ffb7e3da86b12ba5eef2be44 \
    --hash=sha256:d5d69c900b8ca90ff2feb9b89d358abddfbfb2c580359966206b93253a3019a2
```
After these changes the `ubi8-py38` overlay builds successfully.

**Comment by Gregory-Pereira** [comment]
Same changes apply to the `ubi8-py39` overlay as the `f34-py39` overlay.
    1. change [these lines](https://github.com/goern/s2i-thoth/blob/update-churn/ubi8-py39/Dockerfile#L34-L35) to use `/opt/app-root/bin/pip3` instead of `/usr/bin/pip3`.
    2. change [these lines](https://github.com/goern/s2i-thoth/blob/update-churn/ubi8-py39/requirements.txt#L569-L570) to use the correct hashes:
```
    --hash=sha256:8def2131b43f0a6fb157f80dfb277ebfce387155ffb7e3da86b12ba5eef2be44 \
    --hash=sha256:d5d69c900b8ca90ff2feb9b89d358abddfbfb2c580359966206b93253a3019a2
```
After these changes the `ubi8-py39` overlay builds successfully.

**Comment by Gregory-Pereira** [comment]
The changes for the `ubi8-py36` overlay are similar but not exactly the same. You should change [these lines](https://github.com/goern/s2i-thoth/blob/update-churn/ubi8-py36/Dockerfile#L34-L35) to use `/opt/app-root/bin/pip3` instead of `/usr/bin/pip3` as with the over overlays.

However, The difference is that the `thamos` version is set to `1.27.3` instead of `1.27.4`. If this difference in the `thamos` version is intentional skip this step, but if not, additionally alter the hashes for the `thamos` version by changing [these lines](https://github.com/goern/s2i-thoth/blob/update-churn/ubi8-py36/requirements.txt#L569-L570) to use the correct hashes:
```
    --hash=sha256:8def2131b43f0a6fb157f80dfb277ebfce387155ffb7e3da86b12ba5eef2be44 \
    --hash=sha256:d5d69c900b8ca90ff2feb9b89d358abddfbfb2c580359966206b93253a3019a2
```
Either way, after these changes the `ubi8-py36` overlay builds successfully.

**Comment by goern** [comment]
@Gregory-Pereira thanks for the deep look at it!! :100: 
actually, you could contribute these changes to my pr

**Comment by harshad16** [comment]
Thanks, everyone for your contributions and reviews

**Pull request merged**